core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/MetricsServiceImpl.java

```diff
@@ -655,7 +656,16 @@ public void setDefaultTTL(int defaultTTL) {
                         .doOnNext(i -> insertedDataPointEvents.onNext(metric)))
                 .doOnNext(meter::mark);
 
-        Observable<Integer> indexUpdates = dataAccess.updateMetricsIndex(metrics)
+        Observable<Metric<T>> updateIdxMetrics = metrics.filter(metric -> {
```

Comment:
Just to clarify the last comment, the impact on the tests shows that fact that the REST API becomes non-deterministic. Meaning, when a user submits a new metric data point without creating the metric prior the metric is not available for query for a non-deterministic amount of time because there are no guarantees on the state of the metric index write.